src/main/java/edu/harvard/iq/dataverse/api/Dataverses.java

```diff
@@ -342,15 +343,35 @@ public Response setMetadataRoot( @PathParam("identifier")String dvIdtf, String b
     
     @GET
     @Path("{identifier}/facets/")
+    /**
+     * return list of facets for the dataverse with alias `dvIdtf`
+     */
     public Response listFacets( @PathParam("identifier") String dvIdtf ) {
-        return allowCors(response( req -> ok(
-                        execCommand(new ListFacetsCommand(req, findDataverseOrDie(dvIdtf)) )
-                            .stream().map(f->json(f)).collect(toJsonArray()))));
+	    try
+	    {
+		    Dataverse dataverse = findDataverseOrDie(dvIdtf);
+		    JsonArrayBuilder fs = Json.createArrayBuilder();
+		    for( DataverseFacet f : dataverse.getDataverseFacets() )
+		    {
+			    fs.add( f.getDatasetFieldType().getName() );
+		    }
+		    return allowCors( ok( fs ) );
+	    }
+	    catch( WrappedResponse e )
+	    {
+		    return e.getResponse();
+	    }
     }
 
     @POST
     @Path("{identifier}/facets")
     @Produces(MediaType.APPLICATION_JSON)
+    /**
+     * (not publicly documented) API endpoint for assigning facets to a dataverse.
+     * `curl -X POST -H "X-Dataverse-key: $ADMIN_KEY" http://localhost:8088/api/dataverses/$dv/facets --upload-file foo.json`; where foo.json contains a list of datasetField names, 
+     * works as expected (judging by the UI).
+     * This triggers a 500 when '-d @foo.json' is used.
```

Comment:
This looks to me like the usual case where some API's expect form data, and some expect upload-file within the form data - it may make more sense note this for API consistency and defer until there's a systematic check of what endpoints should be getting.